cpp/src/scalar/scalar_factories.cpp

```diff
@@ -165,4 +167,25 @@ std::unique_ptr<scalar> make_default_constructed_scalar(data_type type,
   return type_dispatcher(type, default_scalar_functor{}, stream, mr);
 }
 
+std::unique_ptr<scalar> make_empty_scalar_like(column_view const& column,
+                                               rmm::cuda_stream_view stream,
+                                               rmm::mr::device_memory_resource* mr)
+{
+  std::unique_ptr<scalar> result;
+  switch (column.type().id()) {
+    case type_id::LIST:
+      result = make_list_scalar(empty_like(column)->view(), stream, mr);
+      result->set_valid_async(false, stream);
+      break;
+    case type_id::STRUCT:
+      // Struct scalar inputs must have exactly 1 row.
+      CUDF_EXPECTS(!column.is_empty(), "Can not create empty struct scalar");
+      result = detail::get_element(column, 1, stream, mr);
```

Comment:
index=`1` ? (should be `0` right?)

Reply:
How about `result = make_struct_scalar(column, stream, mr)`?

Reply:
`make_struct_scalar` only accepts  `table_view` or `host_span<column_view>`. In addition, it assumes the size of input data == 1 (rather than slicing them to 1).

Reply:
Wow, expecting `full_size() == 1` (not `sliced_size() == 1`) should be wrong!

Reply:
> index=`1` ? (should be `0` right?)

Yes, I corrected it. Thanks!